ansible/playbooks/adhoc/hotfix/roles/update_master/tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+
+# Default to False before attempting upgrade
+- set_fact: 
+    update_successful:    False
+    downgrade_successful: False
+  when: "{{ update_needed == True }}" 
+
+- name: Update atomic-openshift-master
+  yum:
+    name: "{{ rpm_wanted }}"
+    state: present
+  when: "{{ update_needed == True }}" 
+
+- name: Check if update is successful
+  command: "rpm -qa atomic-openshift"
+  register: post_upgrade
+  changed_when: False
+
+# ansible will return a '17' if a match is found. '-1' means 'not found'
+- name: Set update_successful=True if upgrade succeeded
+  set_fact: 
+    update_successful: True
+  when: "{{ post_upgrade.stdout.find(version_wanted) != -1 }}"
+
+- name: Upgrade failed, so try a downgrade
+  command: "yum -y downgrade atomic-openshift-{{ version_wanted }} atomic-openshift-clients-{{ version_wanted }} atomic-openshift-master-{{ version_wanted }} atomic-openshift-node-{{ version_wanted }} tuned-profiles-atomic-openshift-node-{{ version_wanted }} atomic-openshift-sdn-ovs-{{ version_wanted }}"
+  when: "{{ update_successful == False }}"
```

Comment:
So the reason I like to have it attempt a downgrade is this case:

What if we have nodes with inconsistent versions? I tested this case with this script, and it gets all the nodes running the right version. The result is idempotent because you specify a version you want and it ensures that version gets installed one way or another.

Whereas running upgrade.yml would work on some, but not others, and running downgrade.yml would work on some but not others.

I like having it all-in-one like this, because the human who specifies the version they want will get that version. They don't have to worry about "do I need an upgrade, or a downgrade for this host?". They just request a version, and they get it. So that's what this particular task does, it just ensures the user gets the version they requested, in case the previous task failed.